preprocess_observation_smogon.py

```diff
@@ -65,6 +87,109 @@ def preprocess_observation(I):
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +3, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +4, 0])
+
+            elif 'Misty' in line:
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +0, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +1, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +2, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +3, 1])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +4, 0])
+
+            else:
+                assert('Psychic' in line)
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +0, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +1, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +2, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +3, 0])
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER +4, 1])
+        elif 'sidestart' in line:
+            if 'p1' in line:
+                if ': Spikes' in line:
+                    retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER + NUM_TERRAIN, 1])
+                elif 'Rock' in line:
+                    retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER + NUM_TERRAIN+1, 1])
+                elif 'Toxic' in line:
+                    retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER + NUM_TERRAIN+2, 1])
+                else:
+                    assert('Web' in line)
+                    retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER + NUM_TERRAIN+3, 1])
+            else:
+                assert('p2' in line)
+                if ': Spikes' in line:
+                    retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS*TEAM_SIZE*2 + NUM_STAT_BOOSTS*2 + NUM_WEATHER + NUM_TERRAIN+4, 1])
+                elif 'Rock' in line:
```

Comment:
An object-oriented approach might help make this coding shorter and clearer, e.g. you could have an Attack class that manages this kind of information.

Reply:
I think this issue is going to be resolved at the same time the comment ``A bit of a yellow flag'' gets resolved.